### PR TITLE
Ledger

### PR DESCRIPTION
**Pull request by rehanyahya**

### CODE REVIEW

File: pages/ledger/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const { transactionHistory, error, isLoading } = useLedger();
+  const [totalBalance, setTotalBalance] = useState(0);
+
+  useEffect(() => {
+    if (transactionHistory) {
+      let balance = 0;
+      transactionHistory.data.map((item) => {
+        balance += item.total;
+      });
+      setTotalBalance(balance);
```
Comment (hammadtariq) [inline pages/ledger/index.js:16]: we won’t calculate this on frontend, this should be done on backend

File: pages/ledger/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                  key={`${index}`}
+                >
+                  <div>
+                    <div className={styles.rowDirectionContainer}>
+                      <div className={styles.headingStyle}>Total Balance:</div>
+                      <div className={styles.contentStyle}>{`${item.total.toFixed(2)}`}</div>
+                    </div>
+                    <div className={styles.rowDirectionContainer}>
+                      <div className={styles.headingStyle}>Created At:</div>
+                      <div className={styles.contentStyle}>{`${new Date(item.createdAt).toLocaleString()}`}</div>
```
Comment (hammadtariq) [inline pages/ledger/index.js:46]: no need to add createdAt field

File: pages/api/ledger/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      companyId,
+      userId: user.id,
+      amount: totalAmount,
+      spendType,
+    });
+
+    res.send({
+      message: "Transaction Created",
+      data,
+    });
```
Comment (hammadtariq) [inline pages/api/ledger/index.js:34]:
```suggestion
    res.send(data);
```

File: pages/api/ledger/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      GROUP BY "companies"."id"`,
+      {
+        type: db.Sequelize.QueryTypes.SELECT,
+      }
+    );
+
+    let totalBalance = 0;
+    transactions.map((item) => {
+      totalBalance += item.total;
+    });
```
Comment (hammadtariq) [inline pages/api/ledger/index.js:67]: instead of map use reduce

Comment (rehanyahya) [reply pages/api/ledger/index.js:67]: ok

**Pull request merged**